File: Matlab/test_time_series_AOA.m
fprintf('--------------------------\n')
fprintf('Reading netCDF files \n')
fprintf('--------------------------\n')

tic

var_name = 'aoa';
start_year = 1990;
end_year = 2009;

first_file = 0;

num_days = 0;
coefficient = 365.5;

reference_latitude = -86.0;

reference_directory = '../Data/';

% Loop over the years
for year = start_year:end_year
    %fprintf('Processing files for Year %5g \n', year)

    directory_y = strcat(reference_directory, 'Y', num2str(year), '/');
    list_files = dir(strcat(directory_y, 'runAOA.TR.', num2str(year), '*_1200z.nc4'));

    num_days = num_days + numel(list_files);

    % Loop over the daily files
    for index = 1:numel(list_files)
        filepath = [directory_y,  list_files(index).name];
            
        % Extract information if it is the first file
        if first_file == 0
            longitudes = ncread(filepath, 'lon'); 
            latitudes = ncread(filepath, 'lat'); 
            levels = ncread(filepath, 'lev'); 
            [num_longitudes m] = size(longitudes);
            [num_latitudes m] = size(latitudes);
            [num_levels m] = size(levels);

            lat_index = find(latitudes == reference_latitude);

            %levels = calcPressureLevels(num_levels);
        end;

        % Read the daily average age of air
        var = ncread(filepath, var_name, [1 lat_index 1 1], [Inf 1 Inf 1], [1 1 1 1]) /coefficient; 

        % Determine the zonal mean
        temp_var = mean(var,1);
        temp_var = squeeze(temp_var);

        % Stack the daily values into an existing array
        if first_file == 0
            first_file = 1;
            data_val = temp_var;
        else
            data_val = [data_val, temp_var];
        end
    end
end

% End time
toc

if num_days == 7305
    fprintf('Succesfully processed %5g files \n', num_days)
else
    fprintf('Only processed %5g files. Verify that all is well! \n', num_days)
end

File: Python/test_time_series_AOA.py
#!/usr/bin/env python

##############
# Load modules
##############
from __future__ import print_function

import os
from netCDF4 import Dataset
import numpy as np
import glob
import benchmark_decorator as dectimer

#----------------------------------------
# Function: serial_time_series_processing
#----------------------------------------
@dectimer.bench_time(3)
def serial_time_series_processing():
    """
      Do serial time series processing on a collection of files
    """
    variable_name = 'aoa'
    beginning_year = 1990
    end_year = 2009
    #end_year = 1990

    first_file = 0
    reference_latitude = -86.0

    num_days = 0

    coefficient = 365.5

    reference_directory = '../Data/'

    # Loop over the years
    for year in range(beginning_year, end_year+1):
        #print('Processing year: %d' % (year))
        directory_name = 'Y'+str(year)
        directory_Y = os.path.join(reference_directory, directory_name)

        list_files = glob.glob(directory_Y+"/runAOA.TR."+str(year)+"*_1200z.nc4")

        num_days += len(list_files)

        # Loop over the daily files
        for file in list_files:
            # Open file
            opened_file = Dataset(file, mode='r')

            # Extract information if it is the first file
            if first_file == 0:
                longitudes = opened_file.variables['lon'][:]
                latitudes = opened_file.variables['lat'][:]
                #levels = opened_file.variables['lev'][::-1]
                num_longitudes = np.size(longitudes)
                num_latitudes = np.size(latitudes)
                #num_levels = np.size(levels)

                #levels = pressLevels.calcPressureLevels(num_levels)
                latitude_index = (np.abs(latitudes - reference_latitude)).argmin()
                #print("Latitude index: ", latitude_index)

            # Read the daily average age of air
            age_air = opened_file.variables[variable_name][0, ::-1, latitude_index, :] / coefficient

            # Determine the zonal mean
            zonal_mean = np.mean(age_air, axis=1)

            # Stack the daily values into an existing array
            if first_file == 0:
                first_file = 1
                data_value = zonal_mean
            else:
                data_value = np.column_stack((data_value, zonal_mean))

            # Close file
            opened_file.close()


serial_time_series_processing()

print(' ')
